Added Use of Cone Vision to OneFivePlusBump and OneFiveBumpBalance

avakimc opened the pull request.

avakimc committed:
Added Use of Cone Vision to OneFivePlusBump and OneFiveBumpBalance

tails618 reviewed (changes requested):
Rename "...BumpFirstHalf" to "...Bump" and rename "...BumpSecondHalf" to "...BumpReturn" (Make sure to keep "Red" for the red ones)

balghane commented on src/main/java/frc/robot/commands/autos/OneFiveBalanceBump.java:
```diff
@@ -3,86 +3,103 @@
+                          drive.rotateOrKeepHeading(NORM_SPEED_INTAKING, 0, 0, false, -1);
+                        })
+                    .until(
+                        () -> (drive.getPose().getX() - startXMeters) > DISTANCE_AT_CONE_METERS)),
```
Can we use the new DriveDistance command?

balghane commented on src/main/java/frc/robot/commands/autos/OneFiveBalanceBump.java:
```diff
@@ -3,86 +3,103 @@
+                      intake.stopIntakingGamePiece();
+                    })),
+        new SwerveToPointWrapper(red, drive, new Pose2d (5.85, 2.5, Rotation2d.fromDegrees(0.0)), 2.0),
         new ChargeFarSide(drive, DISTANCE_ONTO_CHARGE_STATION_METERS));
```
Can we use the new `DriveUntilBalanced` in the backwards direction?

balghane commented on src/main/java/frc/robot/commands/autos/OneFivePlusBump.java:
```diff
@@ -0,0 +1,113 @@
+ * component drives to the nearest midfield game object, picks it up, and scores it on the next
+ * nearest cone position.
+ */
+public class OneFivePlusBump extends SequentialCommandGroup {
```
Fix class comment, maybe rename OneFiveBumpReturn?

balghane commented on src/main/java/frc/robot/commands/autos/OneFivePlusBump.java:
```diff
@@ -0,0 +1,113 @@
+        new ParallelDeadlineGroup(
+            new SequentialCommandGroup(
+                new WaitCommand(0.5),
+                new RunCommand(
+                        () -> {
+                          drive.rotateOrKeepHeading(NORM_SPEED_INTAKING, 0, 0, false, -1);
+                        })
+                    .until(
+                        () -> (drive.getPose().getX() - startXMeters) > DISTANCE_AT_CONE_METERS)),
```
I think we need to drive back after the `ParallelDeadlineGroup`, then run the second half trajectory

balghane replied:
Also same comment as the other command about using DriveDistance

balghane commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -101,10 +101,10 @@ public void initialize() {
     autonChooser.addOption(
         "1.5 balance bump blue",
-        new OneFiveBalanceBump(false, false, lift, intake, drive, lights, tagLimelight, scoreLoc));
+        new OneFiveBalanceBump(false, false, lift, intake, drive, lights, intakeLimelight, scoreLoc));
     autonChooser.addOption(
         "1.5 balance bump red",
-        new OneFiveBalanceBump(true, false, lift, intake, drive, lights, tagLimelight, scoreLoc));
+        new OneFiveBalanceBump(true, false, lift, intake, drive, lights, intakeLimelight, scoreLoc));
     autonChooser.addOption(
```
Can you add the two 1.5 bump return autos?

avakimc committed:
Revisions to OneFiveBumpReturn and OneFiveBalanceBump Autos

- changed names of paths for clarity
- changed RunCommands for driving to DriveDistance
- changed ChargeFarSide to DriveUntilBalanced
- Added OneFiveBumpReturn autos to RobotContainer

balghane commented on src/main/java/frc/robot/commands/autos/OneFiveBalanceBump.java:
```diff
@@ -3,86 +3,100 @@
+            }),
+        drive.followTrajectoryCommand(firstTraj, true),
+        // Turn to cone, intake it
+        new InstantCommand(
+            () -> {
+              drive.offsetCurrentHeading(limelight.getAngleToConeDeg());
+            }),
```
I think we need to add one of these to let the robot rotate
```
        new RunCommand(
                () -> {
                  drive.rotateOrKeepHeading(0, 0, 0, true, -1);
                })
            .withTimeout(0.25),
```

balghane commented on src/main/java/frc/robot/commands/autos/OneFiveBalanceBump.java:
```diff
@@ -3,86 +3,100 @@
+        new ParallelDeadlineGroup(
+            new SequentialCommandGroup(
+                new WaitCommand(0.5),
+                new DriveDistance(drive, NORM_SPEED_INTAKING, 1.5, 0.0, false)),
```
Put 1.5 meters into a named variable

balghane commented on src/main/java/frc/robot/commands/autos/OneFiveBalanceBump.java:
```diff
@@ -3,86 +3,100 @@
+                      intake.setDesiredClamped(false);
+                      intake.stopIntakingGamePiece();
+                    })),
+        new SwerveToPointWrapper(red, drive, new Pose2d (5.85, 2.5, Rotation2d.fromDegrees(0.0)), 2.0),
```
Can you make this Pose2d a variable?

balghane commented on src/main/java/frc/robot/commands/autos/OneFiveBumpReturn.java:
```diff
@@ -0,0 +1,109 @@
+public class OneFiveBumpReturn extends SequentialCommandGroup {
+  private static final double DISTANCE_AT_CONE_METERS = 5.9;
+  private static final double NORM_SPEED_INTAKING = 0.3;
+  private double startXMeters = 0;
```
I don't think we need this variable anymore for either of these autos, same for `DISTANCE_AT_CONE_METERS `

balghane commented on src/main/java/frc/robot/commands/autos/OneFiveBumpReturn.java:
```diff
@@ -0,0 +1,109 @@
+            }),
+        drive.followTrajectoryCommand(firstTraj, true),
+        // Turn to cone, intake it
+        new InstantCommand(
+            () -> {
+              drive.offsetCurrentHeading(limelight.getAngleToConeDeg());
+            }),
```
Also need to allow the robot to rotate here

balghane commented on src/main/java/frc/robot/commands/autos/OneFiveBumpReturn.java:
```diff
@@ -0,0 +1,109 @@
+            new SequentialCommandGroup(
+                new WaitCommand(0.5),
+                new DriveDistance(drive, NORM_SPEED_INTAKING, 1.5, 0.0, red),
+                new IntakeSequence(intake, lift, lights)
```
I don't think this parallel deadline group is done right

avakimc committed:
More Revisions to OneFive Autos

- removed unused variables
- fixed parallel deadling groups
- made numbers into variables
- added stuff to make robot rotate to cone

avakimc committed:
Final Revisions

- put rotation command in the right place
